Code/Framework/AzToolsFramework/Tests/Prefab/PrefabUndoDeleteTests.cpp

```diff
@@ -0,0 +1,222 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <Prefab/PrefabTestFixture.h>
+
+#include <AzToolsFramework/Prefab/Undo/PrefabUndoDelete.h>
+#include <AzToolsFramework/Prefab/Undo/PrefabUndoDeleteAsOverride.h>
+
+namespace UnitTest
+{
+    using PrefabUndoDeleteTests = PrefabTestFixture;
+
+    TEST_F(PrefabUndoDeleteTests, PrefabUndoDeleteTests_DeleteEntity)
```

Comment:
For the sake of covering all bases, maybe we should also add a 'PrefabUndoDeleteTests_DeleteNestedInstance' test case?

Reply:
There is this inconsistency in the current workflow that DeleteNestedInstance depends on PrefabUndoInstanceLink to remove the link rather than on new undo nodes.

But I think you're right and it makes sense to have another undo test here for that call. And I will add some comment to describe the inconsistency for reference.